File: src/levihobbs/Models/Scaffold/BookReview.cs
using System;
using System.Collections.Generic;

namespace levihobbs.Models.Scaffold;

public partial class BookReview
{
    public int Id { get; set; }

    public string Title { get; set; } = null!;

    public string AuthorFirstName { get; set; } = null!;

    public string AuthorLastName { get; set; } = null!;

    public int MyRating { get; set; }

    public decimal AverageRating { get; set; }

    public int? NumberOfPages { get; set; }

    public int? OriginalPublicationYear { get; set; }

    public DateTime DateRead { get; set; }

    public string MyReview { get; set; } = null!;

    public virtual ICollection<Bookshelf> Bookshelves { get; set; } = new List<Bookshelf>();
}

File: src/levihobbs/react-apps/book-reviews-app/src/types/BookReview.ts
export interface BookReview {
  id: number;
  title: string;
  authorFirstName: string;
  authorLastName: string;
  titleByAuthor: string;
  myRating: number | null | undefined;
  averageRating: number;
  numberOfPages: number;
  originalPublicationYear: number | null;
  dateRead: string;
  myReview: string;
  searchableString?: string;
  hasReviewContent: boolean;
  previewText: string;
  readingTimeMinutes: number;
  coverImageId?: number | null;
  coverImageUrl?: string | null;
  bookshelves: Bookshelf[];
  toneTags?: string[];
}

export interface Bookshelf {
  id: number;
  name: string;
}

export interface BookshelfGrouping {
  id: number;
  name: string;
  bookshelves: Bookshelf[];
}

export interface BookReviewsViewModel {
  category?: string;
  allBookshelves: Bookshelf[];
  allBookshelfGroupings: BookshelfGrouping[];
  selectedShelf?: string;
  selectedGrouping?: string;
  showRecentOnly: boolean;
  useCustomMappings: boolean;
  bookReviews: BookReview[];
}

export interface BookCoverImage {
  id: number;
  imageData: string;
  fileType: string;
} 

export interface Tag {
  name: string;
  type: 'Genre' | 'Specialty';
  bookshelf?: Bookshelf;
  bookshelfGrouping?: BookshelfGrouping;
}
